llvm/lib/Transforms/Utils/BuildLibCalls.cpp

```diff
@@ -816,7 +816,7 @@ bool llvm::inferNonMandatoryLibFuncAttrs(Function &F,
     Changed |= setDoesNotThrow(F);
     Changed |= setDoesNotCapture(F, 0);
     Changed |= setDoesNotCapture(F, 3);
-    // FIXME: readonly #1?
+    Changed |= setOnlyReadsMemory(F, 0);
```

Comment:
Should the corresponding fread arg be write-only?

Reply:
Right, thanks! Think it should be as well for fgets first arg.